armadillo compilation can fail due to HDF5

https://travis-ci.org/github/CCPPETMR/SIRF-SuperBuild/jobs/662454732?utm_source=github_status
failed on OSX with
```
/Users/travis/build/CCPPETMR/SIRF-SuperBuild/sources/Armadillo/src/wrapper1.cpp:1697:14: error: no matching function for call to 'H5Ovisit3'
      return H5Ovisit(object_id, index_type, order, op, op_data);
             ^~~~~~~~
/usr/local/Cellar/hdf5/1.12.0/include/H5version.h:785:20: note: expanded from macro 'H5Ovisit'
  #define H5Ovisit H5Ovisit3
                   ^~~~~~~~~
/usr/local/Cellar/hdf5/1.12.0/include/H5Opublic.h:211:15: note: candidate function not viable: requires 6 arguments, but 5 were provided
H5_DLL herr_t H5Ovisit3(hid_t obj_id, H5_index_t idx_type, H5_iter_order_t order,
```

Note that HDF5 1.12.0 was installed by brew to install dependencies for `postgis` apparently. Our current version is 1.10 or 1.8. However, this wasn't installed yet as Armadillo doesn't depend on HDF5 in  our project.

I guess that Gadgetron doesn't need Armadillo's HDF5 support, so I suggest to build Armadillo with `-DDETECT_HDF5=OFF`

--- a/SuperBuild/External_Armadillo.cmake
+++ b/SuperBuild/External_Armadillo.cmake
@@ -71,6 +71,7 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
     CMAKE_ARGS
       -DCMAKE_PREFIX_PATH=${SUPERBUILD_INSTALL_DIR}
       -DCMAKE_INSTALL_PREFIX=${${proj}_INSTALL_DIR}
+      -DDETECT_HDF5=OFF
       ${CLANG_ARG}
   )
 

